### ISSUE TITLE
Parsing bugs

### ISSUE THREAD

**Issue by michaelrsweet** [michaelrsweet/mmd#7]
Looks like the markdown parser has some bugs:

- [x] Heading after a block quote is still part of the block quote
- [x] Lines containing a single | are incorrectly interpreted as a table.
- [x] Bulleted lists after preformatted text using indentation are lost.

**Comment by michaelrsweet** [comment]
Table bug seen with the following text:

```
where "-" is a pointer to the sibling node and "|" is a pointer to the first
child or parent node.
```

List bug seen with the following text:

```
The `mxmlGetType` function gets the type of a node:

    mxml_type_t
    mxmlGetType(mxml_node_t *node);

- `MXML_CUSTOM` : A custom value defined by your application,
- `MXML_ELEMENT` : An XML element, CDATA, comment, or processing instruction,
- `MXML_INTEGER` : A whitespace-delimited integer value,
```

**Comment by michaelrsweet** [comment]
Heading bug seem with the following text:

```
Table in block quote:

> Heading 1 | Heading 2 | Heading 3
> --------- | --------- | ---------
> Cell 1,1  | Cell 1,2  | Cell 1,3
> Cell 2,1  | Cell 2,2  | Cell 2,3
> Cell 3,1  | Cell 3,2  | Cell 3,3

# Heading
```

**Comment by michaelrsweet** [comment]
[master e3837f0] Fix some parsing bugs (Issue #7)
